ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/operations/Attestation.java

```diff
@@ -51,7 +52,7 @@ public Attestation(Bitlist aggregation_bits, AttestationData data, BLSSignature
   public Attestation(Attestation attestation) {
     this.aggregation_bits = attestation.getAggregation_bits().copy();
     this.data = attestation.getData();
-    this.signature = BLSSignature.fromBytes(attestation.getAggregate_signature().toBytes());
+    this.signature = attestation.getAggregate_signature();
```

Comment:
Is there no need to copy the signature here? If two attestations refer to the same signature and one is aggregated whilst the other is not, this might cause issues.

Reply:
BLSSignature and all the values it holds are now fully immutable so no need to copy them anywhere. 🎉

Reply:
Noice.